guard/tests/validate.rs

```diff
@@ -328,44 +365,425 @@ mod validate_tests {
+        let mut reader = Reader::new(ReadCursor(Cursor::new(Vec::from(payload.as_bytes()))));
+        let mut writer = Writer::new(WBVec(vec![]), WBVec(vec![]));
+        let status_code = ValidateTestRunner::default()
+            .payload()
+            .run(&mut writer, &mut reader);
+
+        let result = writer.stripped().unwrap();
+        let expected = indoc! {
+            r#"
+            DATA_STDIN[1] Status = FAIL
+            FAILED rules
+            RULES_STDIN[2]/default    FAIL
+            ---
+            Evaluating data DATA_STDIN[1] against rules RULES_STDIN[2]
+            Number of non-compliant resources 0
+            DATA_STDIN[2] Status = FAIL
+            FAILED rules
+            RULES_STDIN[2]/default    FAIL
+            ---
+            Evaluating data DATA_STDIN[2] against rules RULES_STDIN[2]
+            Number of non-compliant resources 0
+            "#
+        };
+
+        assert_eq!(StatusCode::PARSING_ERROR, status_code);
+        assert_eq!(expected, result);
+    }
+
+    #[test]
+    fn test_with_payload_flag_fail_verbose_prev_engine() {
+        let payload = r#"{"data": ["{\"Resources\":{\"NewVolume\":{\"Type\":\"AWS::EC2::Volume\",\"Properties\":{\"Size\":500,\"Encrypted\":false,\"AvailabilityZone\":\"us-west-2b\"}},\"NewVolume2\":{\"Type\":\"AWS::EC2::Volume\",\"Properties\":{\"Size\":50,\"Encrypted\":false,\"AvailabilityZone\":\"us-west-2c\"}}},\"Parameters\":{\"InstanceName\":\"TestInstance\"}}","{\"Resources\":{\"NewVolume\":{\"Type\":\"AWS::EC2::Volume\",\"Properties\":{\"Size\":500,\"Encrypted\":false,\"AvailabilityZone\":\"us-west-2b\"}},\"NewVolume2\":{\"Type\":\"AWS::EC2::Volume\",\"Properties\":{\"Size\":50,\"Encrypted\":false,\"AvailabilityZone\":\"us-west-2c\"}}},\"Parameters\":{\"InstanceName\":\"TestInstance\"}}"], "rules" : [ "Parameters.InstanceName == \"TestInstance\"","Parameters.InstanceName == \"SomeRandomString\"" ]}"#;
+        let mut reader = Reader::new(ReadCursor(Cursor::new(Vec::from(payload.as_bytes()))));
+        let mut writer = Writer::new(WBVec(vec![]), WBVec(vec![]));
+        let status_code = ValidateTestRunner::default()
+            .payload()
+            .previous_engine(true)
+            .verbose(true)
+            .run(&mut writer, &mut reader);
+
+        let result = writer.stripped().unwrap();
+        let expected = indoc! {
```

Comment:
Consider putting this long string into a file and reading from it using `assert_output_from_file_eq` macro.